sympy/solvers/solvers.py

```diff
@@ -2222,7 +2250,10 @@ def update(determined, solution):
             if len(k.free_symbols) != 1:
                 determined[x] = S.Zero
             else:
-                val = solve(k)[0]
+                val = solve(k)
+                if not val:
+                    raise NotImplementedError('assumptions may have prohibited finding a solution')
```

Comment:
Why is `solve` being used here if it potentially leads to this problem?

Reply:
It depends how you define the problem. This can also be viewed as a linear system whose solution is prohibited by assumptions. It were better for the user to have sent it to `diophantine`, but getting a `diophantine` solution from a `solve` call doesn't seem like a less complicated direction to go.

Reply:
I wouldn't have expected this particular `minsolve_linear_system` to be affected by assumptions at all. Also I wouldn't have expected it to use a general purpose function like `solve` at all. Is this not just solving an univariate linear equation?

Reply:
I didn't see the context. I agree that this should not be calling `solve`.